lib/tracing/src/macros.rs

```diff
@@ -0,0 +1,512 @@
+        $crate::__tracing::event!(target: $target, $crate::__tracing::Level::ERROR, { %$($k).+ $($field)* })
+    );
+    (target: $target:expr, $($arg:tt)+ ) => (
+        $crate::__tracing::event!(target: $target, $crate::__tracing::Level::ERROR, {}, $($arg)+)
+    );
+
+    // ...
+    ({ $($field:tt)+ }, $($arg:tt)+ ) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { $($field)+ },
+            $($arg)+
+        )
+    );
+    ($($k:ident).+ = $($field:tt)*) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { $($k).+ = $($field)*}
+        )
+    );
+    (?$($k:ident).+ = $($field:tt)*) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { ?$($k).+ = $($field)*}
+        )
+    );
+    (%$($k:ident).+ = $($field:tt)*) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { %$($k).+ = $($field)*}
+        )
+    );
+    ($($k:ident).+, $($field:tt)*) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { $($k).+, $($field)*}
+        )
+    );
+    (?$($k:ident).+, $($field:tt)*) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { ?$($k).+, $($field)*}
+        )
+    );
+    (%$($k:ident).+, $($field:tt)*) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { %$($k).+, $($field)*}
+        )
+    );
+    (?$($k:ident).+) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { ?$($k).+ }
+        )
+    );
+    (%$($k:ident).+) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { %$($k).+ }
+        )
+    );
+    ($($k:ident).+) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            { $($k).+ }
+        )
+    );
+    ($($arg:tt)+) => (
+        $crate::__tracing::event!(
+            target: $crate::transformed_module_path!(),
+            $crate::__tracing::Level::ERROR,
+            $($arg)+
+        )
+    );
+}
+
+#[macro_export]
+macro_rules! transformed_module_path {
+    () => {
+        $crate::__string_literal_replace!(module_path!() ("zksync_os_" -> "zk::"))
```

Comment:
IMO the complexity is not worth it; also it will be confusing for new devs and they can have issues with finding logs